libexec/rbenv-version-file-read

```diff
@@ -14,6 +14,19 @@ if [ -e "$VERSION_FILE" ]; then
   if [ "$version" = ".." ] || [[ $version == */* ]]; then
     echo "rbenv: invalid version in \`$VERSION_FILE'" >&2
   elif [ -n "$version" ]; then
+    if [[ "$VERSION_FILE" == *\.ruby-version ]]; then
+      variant_file="${VERSION_FILE/\.ruby-version/.ruby-variant}"
```

Comment:
You should explicitly initialize `variant_file` even if the `if` statement is not called, so that random environment variables can't mess with us.